src/DDTrace/SpanContext.php

```diff
@@ -174,6 +126,13 @@ public function getAllBaggageItems()
 
     public function isEqual(SpanContextInterface $spanContext)
     {
+        if (is_subclass_of($spanContext, "DDTrace\Data\SpanContext")) {
+            return $this->traceId == $spanContext->traceId
```

Comment:
Just because PHP's type coercion can get squirrely. :)

```suggestion
            return $this->traceId === $spanContext->traceId
```